File: mqtt/src/main/java/org/sheedon/mqtt/MQTTFactory.java
package org.sheedon.mqtt;

/**
 * mqtt基本处理工厂
 *
 * @Author: sheedon
 * @Email: sheedonsun@163.com
 * @Date: 2020/4/14 20:18
 */
public interface MQTTFactory {

    Call newCall(Request request);

    Observable newObservable(Request request);
}

File: mqtt/src/main/kotlin/org/sheedon/mqtt/MqttFactory.kt
package org.sheedon.mqtt

/**
 * mqtt newCall/newObservable Factory
 * @Author: sheedon
 * @Email: sheedonsun@163.com
 * @Date: 2022/2/13 9:56 下午
 */
interface MqttFactory : CallFactory, ObservableFactory

interface CallFactory {
    fun newCall(request: Request): Call
}

interface ObservableFactory {
    fun newObservable(request: Request): Observable

    fun newObservable(subscribe: Subscribe): Observable
}